src/Sulu/Bundle/SecurityBundle/UserManager/UserManager.php

```diff
@@ -43,13 +110,119 @@ public function __construct(Registry $doctrine, CurrentUserDataInterface $curren
     public function getUserById($id)
     {
         /** @var ObjectRepository $repository */
-        $repository = $this->doctrine->getRepository('SuluSecurityBundle:User');
+        return $this->userRepository->find($id);
+    }
+
+    /**
+     * Deletes a user with the given id
+     *
+     * @param int $id
+     */
+    public function delete($id)
+    {
+        $delete = function ($id) {
+            $user = $this->userRepository->findUserById($id);
+            if (!$user) {
+                throw new EntityNotFoundException(static::$entityName, $id);
+            }
+
+            $this->em->remove($user);
+            $this->em->flush();
+        };
+
+        return $delete;
+    }
+
+    /**
+     * Return all users
+     *
+     * @return array
+     */
+    public function findAll()
+    {
+        return $this->userRepository->findAll();
+    }
+
+    /**
+     * Creates a new user with the given data
+     *
+     * @param array $data
+     *
+     * @return User
+     */
+    public function save(
+        $data,
+        $locale,
+        $id = null,
+        $patch = false
+    ) {
+        $username = $this->getProperty($data, 'username');
+        $contact = $this->getProperty($data, 'contact');
+        $email = $this->getProperty($data, 'email');
+        $password = $this->getProperty($data, 'password');
+        $user = null;
+        try {
+            if ($id) {
+                // PATCH & PUT
+                $user = $this->userRepository->findUserById($id);
+                if (!$user) {
+                    throw new EntityNotFoundException(static::$entityName, $id);
+                }
+                $this->processEmail($user, $email);
+            } else {
+                // POST
+                $user = new User();
+                $this->processEmail($user, $email, $contact);
+            }
+            // check if username is already in database and the current user is not the user with this username
+            if (!$patch || $username !== null) {
+                if ($user->getUsername() != $username &&
+                    !$this->isUsernameUnique($username)) {
+                    throw new UsernameNotUniqueException($username);
+                }
+            }
+            // check if password is valid
+            if (!$patch || $password !== null) {
+                $user->setSalt($this->generateSalt());
+                if ($this->isValidPassword($password)) {
+                    $user->setPassword(
+                        $this->encodePassword($user, $password, $user->getSalt())
+                    );
+                } elseif (!$id) {
+                    throw new MissingPasswordException();
+                }
+            }
```

Comment:
elseif (!$id) {
    // if new user, a password must be set
      throw new MissingPasswordException();
}